packages/table/src/Row.tsx

```diff
@@ -0,0 +1,323 @@
+`;
+
+const displayFlex = css`
+  display: flex;
+  align-items: center;
+`;
+
+const truncation = css`
+  max-width: 100px;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+  overflow: hidden;
+`;
+
+const transitionStyles = {
+  default: css`
+    transition: border 150ms ease-in-out;
+    border-top-color: transparent;
+
+    & > td {
+      padding-top: 0px;
+      padding-bottom: 0px;
+    }
+
+    & > td > ${tdInnerDiv.selector} {
+      max-height: 0;
+    }
+  `,
+
+  entered: css`
+    border-top-color: ${uiColors.gray.light2};
+
+    & > td > ${tdInnerDiv.selector} {
+      max-height: 40px;
+    }
+  `,
+};
+
+function styleColumn(index: string, dataType: DataType) {
+  let justify;
+
+  if (dataType === DataType.Number) {
+    justify = 'flex-end';
+  } else {
+    justify = 'flex-start';
+  }
+
+  return css`
+    & td:nth-child(${index}) > div {
+      justify-content: ${justify};
+    }
+  `;
+}
+
+function getIndentLevelStyle(indentLevel: number) {
+  return css`
+    & > td:nth-child(1) {
+      padding-left: ${8 + indentLevel * 16}px;
+    }
+  `;
+}
+
+const idAllocator = IdAllocator.create('text-input');
+
+interface RowProps extends React.ComponentPropsWithoutRef<'tr'> {
+  expanded?: boolean;
+  disabled?: boolean;
+  indentLevel?: number;
+  isAnyAncestorCollapsed?: boolean;
+}
+
+const Row = React.forwardRef(
+  (
+    {
+      expanded = false,
+      disabled = false,
+      indentLevel = 0,
+      isAnyAncestorCollapsed: isAnyAncestorCollapsedProp,
+      children,
+      className,
+      ...rest
+    }: RowProps,
+    ref: React.Ref<any>,
+  ) => {
+    const {
+      state: { data, columnInfo, hasNestedRows, hasRowSpan },
+      dispatch: tableDispatch,
+    } = useTableContext();
+
+    const indexRef = useRef(idAllocator.generate());
+    const [isExpanded, setIsExpanded] = useState(expanded);
+    const nodeRef = useRef(null);
+
+    useEffect(() => {
+      let shouldDispatchHasNestedRows = false;
+      let shouldDispatchHasRowSpan = false;
+
+      React.Children.forEach(children, child => {
+        if (
+          isComponentType(child, 'Row') &&
+          !shouldDispatchHasNestedRows &&
+          !hasNestedRows
+        ) {
+          shouldDispatchHasNestedRows = true;
+        }
+
+        if (
+          isComponentType(child, 'Cell') &&
+          child.props.rowSpan > 1 &&
+          !hasRowSpan &&
+          !shouldDispatchHasRowSpan
+        ) {
+          shouldDispatchHasRowSpan = true;
+        }
+      });
+
+      if (shouldDispatchHasNestedRows) {
+        tableDispatch({
+          type: TableTypes.SetHasNestedRows,
+          payload: true,
+        });
+      }
+
+      if (shouldDispatchHasRowSpan) {
+        tableDispatch({
+          type: TableTypes.SetHasRowSpan,
+          payload: true,
+        });
+      }
+    }, [children, hasNestedRows, hasRowSpan, tableDispatch]);
+
+    // Iterating over children twice because generated memoized values have different dependants
```

Comment:
This comment is confusing, I see it twice and I think it's more confusing than helpful.

Reply:
Removed it! Had it in because in earlier iterations we were trying to figure out how to combine this map, but the two maps need to update based on different dependents so its more performant to break it up